salt/modules/zypperpkg.py

```diff
@@ -240,6 +242,12 @@ def _is_lock(self):
         """
         return self.exit_code == self.LOCK_EXIT_CODE
 
+    def _is_rpm_lock(self):
+        """
+        Is this an RPM lock error?
+        """
+        return self.RPM_LOCK_ERR_MESSAGE in self.error_msg
```

Comment:
We need to be more strict here to consider the error is actually an RPM lock error.

If we only check if that message exists as part of the error coming from Zypper, we can get the minion into and infitine loop when passing that string as a malicious package name, like:

```
# salt-call pkg.install "can't create transaction lock on"
```

Reply:
With the other locale on the system the message could be different

Reply:
The problem is the exit code from RPM does not specify what kind of error has actually occurred. Exit code is equal to the number of errors of any kind which have happened during the execution.
What we could do here is to simulate RPM behavior and try to acquire the file lock. Would that be the preferred approach?